mycroft/skills/skill_manager.py

```diff
@@ -445,7 +446,11 @@ def handle_converse_request(self, message):
                 try:
                     utterances = message.data['utterances']
                     lang = message.data['lang']
-                    result = skill_loader.instance.converse(utterances, lang)
+                    # check the signature of a converse method to either pass a message or not
+                    if len(signature(skill_loader.instance.converse).parameters) == 3:
+                        result = skill_loader.instance.converse(utterances, lang, message)
```

Comment:
as commented below, can this check instead be made

```python
if len(signature(skill_loader.instance.converse).parameters) == 1:
    result = skill_loader.instance.converse(message)
```

and the converse method in MycroftSkill also changed?

Reply:
sure, on it